code/src/components/product-card/ProductCard.test.tsx

```diff
@@ -54,4 +71,26 @@ describe("ProductCard component", () => {
     const productLink = screen.getByRole("link");
     expect(productLink).toHaveAttribute("href", "/");
   });
+
+  test("should call useState on mount", () => {
+    expect(mockSetState).toHaveBeenCalledWith(true);
+  });
+
+  test("Should call functions on cart icon click", () => {
+    const cartIcon = screen.getByRole("button");
+
+    fireEvent.click(cartIcon);
+
```

Comment:
In  my PR I've added data-testid='add-to-cart-button' to this cart icon to better testing 
Maybe you can also use it to awoid this setState dublication?

Reply:
I added and used 'add-to-cart-button', take a look